pkg/platform/kube/deployer.go

```diff
@@ -250,6 +250,25 @@ func (d *deployer) getFunctionPodLogs(namespace string, name string) (string, st
 			// close the stream
 			logsRequest.Close() // nolint: errcheck
 		}
+
+		eventList, err := d.consumer.kubeClientSet.CoreV1().Events(namespace).List(meta_v1.ListOptions{})
+		if err != nil {
+			d.logger.InfoWith("Failed to get pod events", "err", err)
+			podLogsMessage += "Failed to get pod events\n"
+		}
+
+		var podWarningEvents string
```

Comment:
This become a huge function,
perhaps move it to a different function, return a _list_ of strings so later on its callee can `strings.Join` with its delimiter